students/tammyd/lesson10/mailroom2_working.py

```diff
@@ -0,0 +1,277 @@
+#!/usr/bin/env python
+"""
+mailroom assignment
+This version uses a dict for the main db, and exception handling to
+check input, and has been factored to be amenable to testing.
+"""
+
+import sys
+import math
+
+from functools import reduce
+
+# handy utility to make pretty printing easier
+from textwrap import dedent
+
+
+# In memory representation of the donor database
+# using a tuple for each donor
+# -- kind of like a record in a database table
+# using a dict with a lower case version of the donor's name as the key
+# This makes it easier to have a 'normalized' key.
+#  you could get a bit fancier by having each "record" be a dict, with
+#   "name" and "donations" as keys.
+def get_donor_db():
+    return {'william gates iii': ("William Gates III", [653772.32, 12.17]),
+            'jeff bezos': ("Jeff Bezos", [877.33]),
+            'paul allen': ("Paul Allen", [663.23, 43.87, 1.32]),
+            'mark zuckerberg': ("Mark Zuckerberg", [1663.23, 4300.87, 10432.0]),
+            }
+
+class Donor:
+    def __init__(self, name, donations=None):
+        self.name = name
+        self._donations = [] if donations is None else donations
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def donations(self):
+        return self._donations
+
+    def add_donations(self, donations):
```

Comment:
this looks like untested code --- it has "donations" as a parameter, but is adding "donation" to the list.

you probably want the singular form in both cases.